src/main/java/Command.java

```diff
@@ -0,0 +1,109 @@
+import java.io.IOException;
+
+/**
+ * Handles the different kind of Commands user may have on the chatbot
+ * Generally it is the #Intent of user
+ */
+public class Command {
+
+    public Command() {
+    }
+
+    /**
+     * Running the commands given by user
+     * @param taskList Keep track of all the tasks users have
+     * @param ui Handles the replies given (Dialogue)
+     * @param storage Storing the text files and updating it
+     * @throws DukeException Exception cases
+     * @throws IOException Unexpected exception cases
+     */
+    public void execute(TaskList taskList, UI ui,Storage storage) throws DukeException, IOException {
+        storage.LoadFile();
+        ui.printGreeting();
+        String TaskLine = ui.readCommand();
+        Parser parser = new Parser();
+        outLoop:
```

Comment:
This `execute` method here seems to be the method that drives the entire program -- it contains the while loop that reads the input command and then does the appropriate things with `ui` and `taskList`, depending on what input it is. Might it be better to place this driver loop in the `Duke` class instead?

Also considering that there are really different types of commands (bye, list, done, todo, etc.), perhaps it could be helpful to create subclasses (e.g. `ListCommand`) that inherit from the `Command` class, and implement their own `execute` method (that does the appropriate thing with `ui` and `taskList` depending on what command it is).